compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -1031,6 +1031,28 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                             "the {item_kind} was found for\n{}{}",
                             type_candidates, additional_types
                         ));
+                    } else {
+                        'outer: for inherent_impl_did in self.tcx.inherent_impls(adt.did()) {
+                            for inherent_method in
+                                self.tcx.associated_items(inherent_impl_did).in_definition_order()
+                            {
+                                if let Some(attr) = self.tcx.get_attr(inherent_method.def_id, sym::rustc_confusables)
+                                    && let Some(candidates) = parse_confusables(attr)
+                                    && candidates.contains(&item_name.name)
```

Comment:
I think while it makes sense to have `#[rustc_confusables]` because `#[doc(alias)]` pollutes the search, it still makes sense to actually check *both* for diagnostics, so that we don't have to write 2 annotations when `#[doc(alias)]` is already present

Reply:
I believe #107108 *already* makes the `#[doc(alias)]` check to make suggestions. `#[rustc_confusables]` shouldn't trigger if any `similar_candidate` was present to make the `#[doc(alias)]` "there is a similar method named ..." suggestion.